gengo/languages.yaml

```
@@ -155,6 +158,11 @@ C:
   category: programming
   color: "#88CC88"
   nerd-font-glyph: "\U0000e61d"
+  heuristics:
+    - '(?m)^#include\s+<(iostream|vector|string|algorithm)>'
+    - '(?m)^\s*(namespace\s+\w+\s*\{|using\s+namespace\s+\w+;)'
+    - '(?m)^\s*(class|template|typename)\b'
```

OK, I just checked the heuristics and these two are vulnerable to ReDoS. The issue is that both `^` and `\s` match newlines with the multiline (`?m`) flag. This can cause an exponential increase in steps that the regex engine has to take for some inputs.

So, let's say you have 20 empty lines in a string that fails to match the regex. These would:

1. Check the first line, plus the remaining 19 newlines, fail, and then...
2. Check the second line, plus the remaining 18 newlines, fail, and then...
3. Check the third line, plus the remaining 17 newlines, fail, and then...
4. etc.

These might be less accurate, but are safe from ReDoS. Let me know what you think of these suggestions.

```suggestion
    - '(?m)^(namespace\s+\w+\s*\{|using\s+namespace\s+\w+;)'
    - '(?m)^(class|template|typename)\b'
```